Content improvements (#354)
apple/swift-org-website: index.md

atom: true
---

> <h1 class="preamble">Swift is a <strong>general-purpose</strong> programming language that's <strong>approachable</strong> for newcomers and <strong>powerful</strong> in expert hands. It is <strong>quick</strong> to write, <strong>fast</strong> to run, <strong>modern</strong>, and <strong>safe</strong>.</h1>

Reviewer: Let's go with:
`<h1 class="preamble">Swift is a <strong>general-purpose</strong> programming language that's <strong>approachable</strong> for newcomers and <strong>powerful</strong> for experts. It is <strong>fast</strong>, <strong>modern</strong>, <strong>safe</strong>, and a <strong>joy</strong> to write.`

> <h1 class="preamble">Swift is a <strong>general-purpose</strong> programming language that's <strong>approachable</strong> for newcomers and <strong>powerful</strong> in expert hands. It is <strong>quick</strong> to write, <strong>fast</strong> to run, <strong>modern</strong>, and <strong>safe</strong>.</h1>

Reviewer 2: ```suggestion
<h1 class="preamble">Swift is a <strong>general-purpose</strong> programming language that's <strong>approachable</strong> for newcomers and <strong>powerful</strong> for experts. It is <strong>fast</strong>, <strong>modern</strong>, <strong>safe</strong>, and a <strong>joy</strong> to write.
```

Reviewer 2: cc @daveverwer per @cthielen suggestion ^^

Reviewer 3: I like the change from "in expert hands" to "for experts"

I do prefer "It's quick to write, fast to run, modern, and safe.", but I'm happy if the concensus is otherwise.

{% for snippet in site.data.featured_snippets %}
```swift
{{ snippet -}}
```

> {: class="featured-snippet {% if forloop.first %}visible{% endif %}" }

Reviewer 4: Currently, the height is fixed to not cause visual shifts on refresh. The current code snippets are also just quick examples that haven't been adjusted to take the space into account, which means that the gray space issue is more apparent.

Once we have the actual snippets in place, we should revisit this though to see if we need to adjust anything.

I like the idea of a shuffle button too. 👍

{% endfor %}

<div class="link-grid" markdown="1">
  <ul>
    <li>

>       <a href="/download/#releases">

Reviewer 2: ```suggestion
      <a href="/install">
```

Reviewer 2: @kaishin @shahmishal I think it makes more sense to link to the install page instead of the more complicated download page

Reviewer 5: Sounds good, I accepted the suggestion.

        <div class="flex-container">
          <div class="latest-release-container">
          <span>

>             5.8

Reviewer 2: I was also on the fence, I think we can keep it as is and change later if we need to

>             {{ site.data.builds.swift_releases.last.name }}

Reviewer 4: I changed this button to use the latest available patch version instead of a hard-coded minor version and changed it to link to downloads instead of the latest release blog post (because that won't always be available).

IIRC this is what we wanted to do? Let me know if I should revert that.

          </span>
          </div>
          Latest release
        </div>
      </a>
    </li>

    <li>
      <a href="/getting-started">
        <img src="/assets/images/landing-page/signs.svg" />
        Get started
      </a>
    </li>

    <li>
      <a href="/documentation">
        <img src="/assets/images/landing-page/book.svg" />

>         Read the docs

Reviewer 2: its a good point, but I am not sure which better. it a link to the documentation area which is not directly "learn swift". also "read the docs" is term of art in the context.

>         Read the docs

Reviewer 2: @tkremenek @airspeedswift opinions?

      </a>
    </li>
  </ul>
</div>

<div class="links links-list-nostyle" markdown="1">

>   - [Learn more about Swift](/about)

Reviewer 2: @daveverwer @dempseyatgithub opinions?

>   - [Learn more about Swift](/about)

Reviewer 2: I think we should remove this link

</div>

> ## Use cases

Reviewer 4: I have fixed this to keep things consistent for now. Once we start the redesign project we can revisit whether we want to keep this casing or not.

> <div class="links links-list-nostyle" markdown="1">
>   - [Learn more about Swift](/about)
> </div>

Reviewer 2: ```suggestion
```

Reviewer 2: @daveverwer @dempseyatgithub per @cthielen suggestion I think we can remove this link

<ul class="use-cases">
  <li>

>     <h3>Apple platforms</h3>

Reviewer: Let's do "Apple Platforms" to keep the casing consistent with "Server and Networking".

>     <h3>Apple platforms</h3>

Reviewer 2: ```suggestion
    <h3>Apple Platforms</h3>
```

    <p>

>       Swift is a powerful and intuitive programming language optimized when running on iOS, macOS, and other Apple platforms.

Reviewer 2: @tkremenek @airspeedswift opinions?

      <br><br>
      Apple offers a wide variety of frameworks and APIs that make applications developed for these platforms unique and fun.
    </p>
    <a href="https://developer.apple.com/swift/resources/" class="cta-secondary">Learn more</a>
  </li>
  <li>
    <h3>Cross-platform Command-line</h3>
    <p>

>       Writing Swift is interactive and fun, the syntax is concise yet expressive.

Reviewer 2: @tkremenek @airspeedswift opinions?

      Swift code is safe by design and produces software that runs lightning-fast.
      <br><br>

>       SwiftArgumentParser and Swift's growing package ecosystem make developing cross-platform command-line tools a breeze.

Reviewer 2: @cthielen the goal here was to put a light on Swift's package ecosystem in addition to the language itself

>       SwiftArgumentParser and Swift's growing package ecosystem make developing cross-platform command-line tools a breeze.

Reviewer 2: @tkremenek @airspeedswift opinions?

    </p>

    <a href="/getting-started/cli-swiftpm" class="cta-secondary">Learn more</a>
  </li>
  <li>

>     <h3>Server and Networking</h3>

Reviewer 2: this is intentional, networking applications are more than just web services, and server is more than just networking. that said, I dont think we need to treat them as separate use cases as they are related

>     <h3>Server and Networking</h3>

Reviewer 2: @tkremenek @airspeedswift opinions?

    <p>
      Swift's small memory footprint, quick startup time, and deterministic performance make it a great choice for server and other networked applications.
      <br><br>

>       SwiftNIO and Swift's dynamic server ecosystem bring joy to developing networked applications.

Reviewer 2: @cthielen the goal here was to put a light on Swift's package ecosystem in addition to the language itself

    </p>

    <a href="/server" class="cta-secondary">Learn more</a>
  </li>
</ul>

## Getting involved

> Everyone is welcome to contribute to Swift. Contributing doesn’t just mean submitting pull requests—there are many different ways for you to get involved, including answering questions on the forums, reporting or triaging bugs, and participating in the Swift evolution process.

Reviewer 2: ```suggestion
Everyone is welcome to contribute to Swift. Contributing doesn’t just mean wiring code or submitting pull request — there are many different ways for you to get involved, including answering questions on the forums, reporting or triaging bugs, and participating in the Swift evolution process.
```

Reviewer 6: I think "wiring" in the rewrite should be "writing".

> Everyone is welcome to contribute to Swift. Contributing doesn’t just mean wiring code or submitting pull request — there are many different ways for you to get involved, including answering questions on the forums, reporting or triaging bugs, and participating in the Swift evolution process.

Reviewer 2: ```suggestion
Everyone is welcome to contribute to Swift. Contributing doesn’t just mean writing code or submitting pull request — there are many different ways for you to get involved, including answering questions on the forums, reporting or triaging bugs, and participating in the Swift evolution process.
```

<ul class="getting-involved">
  <li>
    <h3>Design</h3>
    <p>
      Help shape the future of Swift by participating in <em>the Swift evolution process</em>.
    </p>
    <a href="/contributing/#swift-evolution" class="cta-secondary">Learn more</a>
  </li>
  <li>
    <h3>Code</h3>
    <p>
      Contribute to the Swift compiler, standard library, and other core components of the project.
    </p>
    <a href="/contributing/#contributing-code" class="cta-secondary">Learn more</a>
  </li>
  <li>
    <h3>Troubleshoot</h3>
    <p>
      Help improve the quality of Swift by reporting and triaging bugs.
    </p>
    <a href="/contributing/#triaging-bugs" class="cta-secondary">Learn more</a>
  </li>
</ul>

## What's new

Stay up-to-date with the latest in the Swift community.

<div class="links links-list-nostyle" markdown="1">
  - [Visit the Swift.org blog](/blog/)
  - [Visit the Swift forums](https://forums.swift.org)

>   - [Follow @Swiftlang on Twitter](https://twitter.com/swiftlang){:target="_blank" class="link-external"}

Reviewer 7: ```suggestion
  - [Follow @Swiftlang on X](https://twitter.com/swiftlang){:target="_blank" class="link-external"}
```

>   - [Follow @Swiftlang on Twitter](https://twitter.com/swiftlang){:target="_blank" class="link-external"}

Reviewer 2: ```suggestion
  - [Follow @swiftlang on X (formally Twitter)](https://twitter.com/swiftlang){:target="_blank" class="link-external"}
```

Reviewer 6: The rewrite should be "formerly Twitter", not "formally".

>   - [Follow @swiftlang on X (formally Twitter)](https://twitter.com/swiftlang){:target="_blank" class="link-external"}

Reviewer 2: ```suggestion
  - [Follow @swiftlang on X (formerly Twitter)](https://twitter.com/swiftlang){:target="_blank" class="link-external"}
```
